packages/@aws-cdk/core-synthesizer/lib/staging-stack.ts

```diff
@@ -0,0 +1,152 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { DockerImageAssetSource, RemovalPolicy, Stack, StackProps } from '@aws-cdk/core';
+import { Construct, IConstruct } from 'constructs';
+
+/**
+ * Information on how a Staging Stack should look.
+ */
+export interface IStagingStack extends IConstruct {
+  /**
+   * The app-scoped, environment-keyed bucket created in this staging stack.
+   */
+  readonly stagingBucket: s3.Bucket;
+
+  /**
+   * The well-known name of the app-scpoed, environment-keyed bucket created in this staging stack.
+   */
+  readonly stagingBucketName: string;
+
+  /**
+   * The app-scoped, environment-keyed repositories created in this staging stack.
+   * A repository is created per image asset family.
+   */
+  readonly stagingRepos: Record<string, ecr.Repository>;
+
+  /**
+   * The well-known name of the IAM role assumed for publishing to s3.
+   */
+  readonly fileAssetPublishingRoleArn: string;
+
+  /**
+   * The well-known name of the IAM role assumed for publishing to ecr.
+   */
+  readonly dockerAssetPublishingRoleArn: string;
+
+  /**
+   * Returns the well-known name of the app-scoped, environment-keyed repository associated
+   * with the given asset.
+   */
+  getRepoName(asset: DockerImageAssetSource): string;
+}
+
+/**
+ * Staging Stack Properties
+ */
+export interface StagingStackProps extends StackProps {
+  /**
+   * Explicit name for the staging bucket
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly stagingBucketName?: string;
+
+  /**
+   * Pass in an existing role to be used as the file publishing role.
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly fileAssetPublishingRoleArn?: string;
+
+  /**
+   * Pass in an existing role to be used as the image publishing role.
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly dockerAssetPublishingRoleArn?: string;
+}
+
+/**
+ * A default Staging Stack
+ */
+export class StagingStack extends Stack implements IStagingStack {
+  /**
+   * Default asset publishing role ARN for file (S3) assets.
+   */
+  public static readonly DEFAULT_FILE_ASSET_PUBLISHING_ROLE_ARN = 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/cdk-${Qualifier}-file-publishing-role-${AWS::AccountId}-${AWS::Region}';
+
+  /**
+   * Default asset publishing role ARN for docker (ECR) assets.
+   */
+  public static readonly DEFAULT_DOCKER_ASSET_PUBISHING_ROLE_ARN = 'cdk-${Qualifier}-asset-publishing-role-${AWS::AccountId}-${AWS::Region}';
```

Comment:
I'm not entirely sure how these parameters work and how I can use them in the staging stack. for example, right now the s3 bucket is called `default-bucket` because I can't add something like `cdk-${AWS::AccountId}-${AWS::Region}` into the name.

Reply:
Why not?

```ts
new Bucket(this, 'Bucket', {
  bucketName: `cdk-${this.qualifier}-${this.region}-${this.account]`,
});
```

?